lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -1021,7 +1021,7 @@ private function formatType(
         \ReflectionMethod $method,
         \ReflectionParameter $parameter = null
     ) {
-        $name      = (string) $type;
+        $name      = $type->getName();
```

Comment:
The generated string for a nullable type must be `?\Foo`, not just `\Foo`

Reply:
I think this part handle nullable already https://github.com/doctrine/common/blob/93479a34af73b55a7ac085315c62ca73b135ceeb/lib/Doctrine/Common/Proxy/ProxyGenerator.php#L1054-L1058

Reply:
Hmm, yep, patch seems good as-is 👍